execution-engine/src/fs.rs

```diff
@@ -912,7 +920,7 @@ impl FileSystem {
     #[inline]
     pub(crate) fn random_get(&self, buf_len: Size) -> FileSystemResult<Vec<u8>> {
         let mut buf = vec![0; buf_len as usize];
-        if let result::Result::Success = getrandom(&mut buf) {
+        if let result::Result::Success(_) = getrandom(&mut buf) {
```

Comment:
This is ugly: I'd define a `is_success()` method on the `result::Result` type and use `if getrandom(&mut but).is_success()` instead.  This is also more in line with idiomatic Rust patterns for similar types in the Rust standard library (e.g. `Result` and `Option`).

Reply:
The crate is labelled with no_std. We could weaken this limitation.

Reply:
@ShaleXIONG Interesting. What do you suggest? What features of `std` do we need here?